src/bridging/providers/across/util.ts

```diff
@@ -29,43 +37,117 @@ export function getTokenAddress(tokenSymbol: string, chainConfig: AcrossChainCon
   return chainConfig.tokens[tokenSymbol]
 }
 
-export function toBridgeQuoteResult(amount: string, suggestedFees: SuggestedFeesResponse): AcrossQuoteResult {
-  // TODO: Do we need to use all the other fees, or they are included in totalRelayFee? I know 'lpFee' fee is, as stated in the docs, but not sure about the others.
+export function toBridgeQuoteResult(
+  request: QuoteBridgeRequest,
+  slippageBps: number,
+  suggestedFees: SuggestedFeesResponse
+): AcrossQuoteResult {
+  const { kind } = request
+
+  return {
+    isSell: kind === OrderKind.SELL,
+    amountsAndCosts: toAmountsAndCosts(request, slippageBps, suggestedFees),
+    quoteTimestamp: Number(suggestedFees.timestamp),
+    expectedFillTimeSeconds: Number(suggestedFees.expectedFillTimeSec),
+    suggestedFees,
+  }
+}
+
+function toAmountsAndCosts(
+  request: QuoteBridgeRequest,
+  slippageBps: number,
+  suggestedFees: SuggestedFeesResponse
+): BridgeQuoteAmountsAndCosts {
+  const { amount, sellTokenDecimals, buyTokenDecimals } = request
+
+  // Get the amounts before fees
+  const sellAmountBeforeFeeBig = getBigNumber(amount, sellTokenDecimals)
+  const sellAmountBeforeFee = sellAmountBeforeFeeBig.big
+  const buyAmountBeforeFee = getBigNumber(sellAmountBeforeFeeBig.num, buyTokenDecimals).big // Sell and buy token should be the same asset for current implementation, but technically they can have different decimals
+
+  // Apply the fee to the buy amount (sell amount doesn't change)
+  const totalRelayerFeePct = BigInt(suggestedFees.totalRelayFee.pct)
+  const buyAmountAfterFee = applyPctFee(buyAmountBeforeFee, totalRelayerFeePct)
+
+  // Calculate the fee
+  const feeSellToken = sellAmountBeforeFee - applyPctFee(sellAmountBeforeFee, totalRelayerFeePct)
+  const feeBuyToken = buyAmountBeforeFee - buyAmountAfterFee
```

Comment:
To make sure I understand: bridge sell and buy token and amounts are expected to be exactly the same, although they can have different decimals for some reason.
Because of the different decimals, the fee is applied to both buy and sell amounts. 
Is that right?